Tests/src/PublicAPITests/SystemResultTests.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Awaitable.h"
+#include "CSP/Common/Array.h"
+#include "CSP/Multiplayer/MultiPlayerConnection.h"
+#include "CSP/Systems/Assets/AssetSystem.h"
+#include "CSP/Systems/Spaces/SpaceSystem.h"
+#include "CSP/Systems/SystemsManager.h"
+#include "CSP/Web/HTTPResponseCodes.h"
+#include "Services/ApiBase/ApiBase.h"
+#include "SpaceSystemTestHelpers.h"
+#include "TestHelpers.h"
+#include "UserSystemTestHelpers.h"
+
+#include "gtest/gtest.h"
+#include <CallHelpers.h>
+#include <Web/HttpRequest.h>
+#include <Web/WebClient.h>
+
+#ifdef CSP_WASM
+	#include "Web/EmscriptenWebClient/EmscriptenWebClient.h"
+#else
+	#include "Web/POCOWebClient/POCOWebClient.h"
+#endif
+
+
+using namespace csp::common;
+using namespace csp::systems;
+
+namespace SystemResultsTest
+{
+
+typedef std::function<void(const NullResult& Result)> NullResultCallback;
+
+#ifdef CSP_WASM
+
+class TestWebClient : public csp::web::EmscriptenWebClient
+{
+public:
+	TestWebClient(const csp::web::Port InPort, const csp::web::ETransferProtocol Tp) : EmscriptenWebClient(InPort, Tp, false)
+	{
+	}
+};
+
+#else
+
+class TestWebClient : public csp::web::POCOWebClient
+{
+public:
+	TestWebClient(const csp::web::Port InPort, const csp::web::ETransferProtocol Tp) : POCOWebClient(InPort, Tp, false)
+	{
+	}
+};
+
+#endif
+
+class ResponseReceiver : public ResponseWaiter, public csp::web::IHttpResponseHandler
+{
+public:
+	ResponseReceiver() : ResponseReceived(false), ThreadId(std::this_thread::get_id())
+	{
+	}
+
+	void OnHttpResponse(csp::web::HttpResponse& InResponse) override
+	{
+		// We expect the callback to have come from a seperate Thread
+		// EXPECT_FALSE(std::this_thread::get_id() == ThreadId);
```

Comment:
Can we either remove this commented-out line, or (even better) uncomment it and validate the assertion?